libparsec/crates/platform_device_loader/src/web/error.rs

```diff
@@ -0,0 +1,205 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use libparsec_types::{anyhow, thiserror};
+use web_sys::wasm_bindgen::JsValue;
+
+#[derive(Debug, thiserror::Error)]
+pub enum NewStorageError {
+    #[error("No window context available, are we in a browser?")]
+    NoWindow,
+    #[error("No local storage available")]
+    NoLocalStorage,
+    #[error("Failed to access local storage ({:?})", .0)]
+    WindowError(JsValue),
+}
+
+macro_rules! impl_from_new_storage_error {
+    ($for:ty) => {
+        impl From<NewStorageError> for $for {
+            fn from(value: NewStorageError) -> Self {
+                Self::Internal(anyhow::anyhow!("{value}"))
+            }
+        }
+    };
+    ($($for:ty),*) => {
+        $(impl_from_new_storage_error!($for);)*
+    };
+}
+
+impl_from_new_storage_error!(
+    crate::LoadDeviceError,
+    crate::SaveDeviceError,
+    crate::ChangeAuthentificationError,
+    crate::ArchiveDeviceError,
+    crate::RemoveDeviceError
+);
+
+error_set::error_set! {
+    GetItemStorageError = {
+        #[display("Failed to get item {key} from storage ({error:?})")]
+        GetItemStorage {
+            key: String,
+            error: JsValue
+        }
+    };
+    SetItemStorageError = {
+        #[display("Failed to set item {key} to storage ({error:?})")]
+        SetItemStorage {
+            key: String,
+            error: JsValue
+        }
+    };
+    RemoveItemStorageError = {
+        #[display("Failed to remove item {key} from storage ({error:?})")]
+        RemoveItemStorage {
+            key: String,
+            error: JsValue
+        }
+    };
+    JsonDeserializationError = {
+        JsonDecode(serde_json::Error)
+    };
+    JsonSerError = {
```

Comment:
In the codebase we consider the encoding is always possible (since we encode only valid data), so there is no need for this error (we should just panic if it occurs)

see what is done in the `libparsec_types::format_v0_dump` function

Reply:
done